packages/core/useClickOutside/index.ts

```diff
@@ -0,0 +1,55 @@
+import { Fn, isClient, MaybeRef, tryOnUnmounted } from '@vueuse/shared'
+import { ref, watch } from 'vue-demi'
+import { useEventListener } from '../useEventListener'
+
+/**
+ * Listen for clicks outside of an element.
+ *
+ * @see   {@link https://vueuse.js.org/useClickOutside}
+ * @param target
+ * @param handler
+ */
+
+const events = ['mousedown', 'touchstart'] as const
+
+export function useClickOutside(
+  target: MaybeRef<Element | null | undefined>,
+  handler: EventListener,
+) {
+  if (!isClient)
+    return
+
+  const targetRef = ref(target)
+
+  const listener = (event: MouseEvent | TouchEvent) => {
+    const elements = event.composedPath()
+    if (targetRef.value === event.target || elements.includes(targetRef.value!))
+      return
+
+    handler(event)
+  }
+
+  let listeners: Fn[] = []
+
+  const stopListeners = () => {
+    listeners.forEach(stop => stop())
+    listeners = []
+  }
+
+  tryOnUnmounted(stopListeners)
+
+  watch(
+    targetRef,
+    () => {
+      if (!targetRef.value) {
+        stopListeners()
```

Comment:
```suggestion
      stopListeners()
      if (!targetRef.value) {
```

Should we unbind the last element when the element ref changed?

Reply:
Wait, since you are listening on window not the element, maybe there is no need to register events inside the watch, just do it outside will be enough.